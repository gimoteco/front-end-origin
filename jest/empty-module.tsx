import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
