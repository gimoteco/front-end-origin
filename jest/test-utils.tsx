import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
