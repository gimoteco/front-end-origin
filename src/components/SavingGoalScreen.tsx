import * as React from 'react';
import Fade from 'react-reveal/Fade';
import HouseIcon from '../icons/house.svg';
import { calculateMonthlyDeposit } from '../utils/format';
import { parseDateParts } from '../utils/parser';
import BaseLayout from './BaseLayout';
import Button from './Button';
import Card from './Card';
import DepositsInfoMessage from './DepositsInfoMessage';
import Field from './Field';
import { InputType } from './Input';
import MonthlyValue from './MonthlyValue';
import * as Style from './SavingGoalScreen.styles';
import Segment from './Segment';

function SavingGoalScreen() {
  const [total, setTotal] = React.useState(null);
  const [reachDate, setReachDate] = React.useState(new Date());

  const { numberOfDeposits, depositValue } = calculateMonthlyDeposit(
    total,
    reachDate
  );
  const { month, year } = parseDateParts(reachDate);

  return (
    <BaseLayout>
      <Fade>
        <Style.Instruction>
          Let's plan your <strong>saving goal.</strong>
        </Style.Instruction>
      </Fade>

      <Card title="Buy a house" subtitle="Saving goal" icon={<HouseIcon />}>
        <form>
          <Style.FormFieldsWrapper>
            <Field
              id="totalAmount"
              label="Total amount"
              value={total}
              type={InputType.Money}
              onChange={(newValue: number) => setTotal(newValue)}
            />

            <Field
              id="reachDate"
              label="Reach goal by"
              value={reachDate}
              type={InputType.Month}
              onChange={(newValue: Date) => setReachDate(newValue)}
            />

            <Segment>
              {{
                top: <MonthlyValue value={depositValue} />,
                bottom: (
                  <DepositsInfoMessage
                    {...{ numberOfDeposits, total, month, year }}
                  />
                )
              }}
            </Segment>
          </Style.FormFieldsWrapper>

          <Button>Confirm</Button>
        </form>
      </Card>
    </BaseLayout>
  );
}

export default React.memo(SavingGoalScreen);
