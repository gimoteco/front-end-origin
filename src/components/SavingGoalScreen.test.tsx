import React from 'react';
import { fireEvent, render } from '../../jest/test-utils';
import SavingGoalScreen from './SavingGoalScreen';

describe('SavingGoalScreen', () => {
  it('should render according snapshot', () => {
    const { container } = render(<SavingGoalScreen />);

    expect(container).toMatchSnapshot();
  });

  it('should compute the deposit value', async () => {
    const { getByTestId, findAllByRole, findByText } = render(
      <SavingGoalScreen />
    );
    const totalInput = getByTestId('totalAmount');

    fireEvent.change(totalInput, {
      target: { value: 1000 }
    });
    fireEvent.click((await findAllByRole('button'))[1]);

    expect(await findByText('$500')).toBeInTheDocument();
  });
});
