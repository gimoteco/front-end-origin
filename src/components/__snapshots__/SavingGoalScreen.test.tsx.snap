// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavingGoalScreen should render according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa hFUtUI"
  >
    <header
      class="sc-htpNat fQvjWA"
    >
      <div
        class="react-reveal"
      >
        <span />
      </div>
    </header>
    <main
      class="sc-bwzfXH hEEQmz"
    >
      <p
        class="sc-kGXeez dlOmRw instruction"
      >
        Let's plan your 
        <strong>
          saving goal.
        </strong>
      </p>
      <div
        class="sc-ifAKCX bzTDQq"
      >
        <div
          class="sc-gzVnrw liXRWx"
        >
          <span />
          <h1
            class="sc-EHOje eTDScl"
          >
            Buy a house
          </h1>
          <h2
            class="sc-bZQynM Taavd"
          >
            Saving goal
          </h2>
        </div>
        <form
          class="sc-ckVGcZ hLslGA"
        >
          <div
            class="sc-kpOJdX ejQcxA"
          >
            <div
              class="sc-dxgOiQ jckyxV"
            >
              <div
                class="sc-htoDjs bAnpk"
              >
                <label
                  class="sc-dnqmqq bkKgUr"
                  for="totalAmount"
                >
                  Total amount
                </label>
                <div
                  class="sc-iwsKbI iTofTH"
                >
                  <label
                    class="sc-gqjmRU jUvAWB"
                    for="totalAmount"
                  >
                    $
                  </label>
                  <input
                    class="sc-gZMcBi ipPELa"
                    data-testid="totalAmount"
                    id="totalAmount"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="sc-htoDjs bAnpk"
              >
                <label
                  class="sc-dnqmqq bkKgUr"
                  for="reachDate"
                >
                  Reach goal by
                </label>
                <div
                  class="sc-jTzLTM dQdxiU"
                  data-testid="reachDate"
                  id="reachDate"
                  tabindex="0"
                >
                  <button
                    class="sc-VigVT cQoHTC"
                    disabled=""
                    type="button"
                  >
                    <span />
                  </button>
                  <div
                    class="sc-fjdhpX jKSagH"
                  >
                    <span
                      class="sc-jzJRlG kFLwJY"
                    >
                      May
                    </span>
                    <span
                      class="sc-cSHVUG zqSoN"
                    >
                      2020
                    </span>
                  </div>
                  <button
                    class="sc-VigVT cQoHTC"
                    type="button"
                  >
                    <span />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="sc-jKJlTe kdnoMZ"
            >
              <div
                class="sc-eNQAEJ ckPgud"
              >
                <div
                  class="sc-kAzzGY ftdBxt"
                >
                  <span
                    class="sc-chPdSV kMTFUo"
                  >
                    Monthly
                  </span>
                  <span
                    class="sc-kgoBCf bjhxvA"
                  >
                    $0
                  </span>
                </div>
              </div>
              <div
                class="sc-hMqMXs nVhyb"
              >
                <p>
                  Youâ€™re planning 
                  <strong>
                    1
                     monthly deposits
                  </strong>
                   to reach your 
                  <strong>
                    $0
                  </strong>
                   goal by
                   
                  <strong>
                    May
                     
                    2020
                    .
                  </strong>
                </p>
              </div>
            </div>
          </div>
          <button
            class="sc-bxivhb bEbvQw"
            type="button"
          >
            Confirm
          </button>
        </form>
      </div>
    </main>
  </div>
</div>
`;
